helpers/GenerateResourcesAndImage.ps1

```diff
@@ -115,7 +130,13 @@ Function GenerateResourcesAndImage {
     $ServicePrincipalClientSecret = $env:UserName + [System.GUID]::NewGuid().ToString().ToUpper();
     $InstallPassword = $env:UserName + [System.GUID]::NewGuid().ToString().ToUpper();
 
-    Connect-AzAccount
+    if (([string]::IsNullOrEmpty($AzureClientId)))
+    {
+        Connect-AzAccount
+    } else {
+        $AzureAppCred = (New-Object System.Management.Automation.PSCredential $AzureClientId, (ConvertTo-SecureString $AzureClientSecret -AsPlainText -Force))
+        Connect-AzAccount -ServicePrincipal -Credential $AzureAppCred -Tenant $AzureTenantId
```

Comment:
I'd suggest the following for more visibility:
```
$AzSecureSecret = ConvertTo-SecureString $AzureClientSecret -AsPlainText -Force
$AzureAppCred = New-Object System.Management.Automation.PSCredential($AzureClientId, $AzSecureSecret)
Connect-AzAccount -ServicePrincipal -Credential $AzureAppCred -Tenant $AzureTenantId
```

Reply:
Looks much better and legible; thanks for the feedback. I have updated the code with your suggestion.